API_202404/src/main/java/com/example/api_202404/controllers/SchoolController.java

```diff
@@ -16,23 +20,39 @@ public class SchoolController {
 
     private final SchoolService schoolService;
 
+
     @GetMapping
-    public List<School> getAllSchools() {
-        return schoolService.findAllSchools();
+    public List<SchoolDTOOutput> getAllSchools() {
+        return SchoolConverter.convertEntityListToDTO(schoolService.findAllSchools());
     }
 
-    @GetMapping(value = "/{id}")
-    public School getSchoolById(@PathVariable Long id) {
-        return this.schoolService.findSchoolById(id);
+    @GetMapping(value = "/{schoolId}")
+    public SchoolDTOOutput getSchoolById(@PathVariable Long schoolId) {
+        return SchoolConverter.convertEntityToDTO(this.schoolService.findSchoolById(schoolId));
     }
 
-    @GetMapping(value = "/students/{schoolId}")
-    public List<Student> getAllStudentsBySchoolId(@PathVariable Long schoolId) {
-        return this.schoolService.findAllStudentsBySchoolId(schoolId);
+    @GetMapping(value = "/{schoolId}/students")
+    public List<StudentDTOOutput> getAllStudentsBySchoolId(@PathVariable Long schoolId) {
+        return StudentConverter.convertEntityListToDTO(this.schoolService.findAllStudentsBySchoolId(schoolId));
     }
 
     @GetMapping(value = "/{schoolId}/students/{studentId}")
-    public Student getStudentBySchoolIdAndStudentId(@PathVariable Long schoolId, @PathVariable Long studentId) {
-        return this.schoolService.findStudentBySchoolIdAndStudentId(schoolId, studentId);
+    public StudentDTOOutput getStudentBySchoolIdAndStudentId(@PathVariable Long schoolId, @PathVariable Long studentId) {
+        return StudentConverter.convertEntityToDTO(this.schoolService.findStudentBySchoolIdAndStudentId(schoolId, studentId));
+    }
+
+    @PostMapping
+    public String addSchool(@RequestBody SchoolDTOInput schoolDTOInput) {
+        return "New school added:\n" + SchoolConverter.convertEntityToDTO(schoolService.addSchool(SchoolConverter.convertDTOToEntity(schoolDTOInput)));
```

Comment:
Sakyčiau visuose responsuose galim daryti du variantus - custom message (gal su kokiu objekto ID ar pan.), kad "kažkas sėkmingai pridėta", arba json to objekto kurį pridėjom, šiuo atveju - yra custom message + objekto json, kas yra = per ilgai custom žinutei, ir taip pat != normaliam json'ui kurį front-endas galėtų parsinti, tai rekomenduočiau arba tą arba tą daryti:)